plugins/fritz-box7490/de/fritzbox7490.sh

```diff
@@ -0,0 +1,204 @@
+		SID=$($WEBCLIENT "$FritzBoxURL$FRITZLOGIN$GETDATA" | sed -n '/.*<SID>\([^<]*\)<.*/s//\1/p')
+		if $Debug; then echo "****************** Logged in with SID=$SID" >> $LOGFILE; fi
+	else
+		SID=$SessionInfoSID
+		if $Debug; then echo "****************** LOGIN: Bereits erfolgreiche SID: $SID" >> $LOGFILE; fi
+	fi
+	if [ "$SID" = "0000000000000000" ]; then
+		if $Debug; then echo "****************** LOGIN: ERROR - Konnte keine gueltige SID ermitteln" >> $LOGFILE; fi
+	fi
+  }
+
+# Definition der Linienlänge für die Ausgabe (MEZ oder MESZ?)
+# Definition of line length for output (CET or CEST?)
+  MESZ=$(echo $(date +"%Z") | awk '{print length($0)-3}')
+  line="#####################################################"
+  if [ "$MESZ" = "1" ]; then line="${line}#";fi
+
+# Die alte Log-Datei für Differenzbetrachtungen sichern
+# Save the old log file for differential views
+  if ! $Debug; then mv ${LOGFILE} ${LOGFILE}.1; fi
+
+# Startmeldung zur Ausgabe
+# Start message for output
+  echo $line                                                      > $LOGFILE
+  echo "Startzeit der Auswertung: $(date +"%d.%m.%Y %X %Z")   #" >> $LOGFILE
+  echo $line                                                     >> $LOGFILE
+  echo                                                           >> $LOGFILE
+
+# Abrage der Hard-/Software-Eigenschaften, funktioniert ohne Anmeldung!
+# Query the hardware / software properties, works without login! 
+  jBoxInfo=$($WEBCLIENT "$FritzBoxURL/jason_boxinfo.xml")
+  fbName=$(echo "$jBoxInfo" | grep 'j:Name' | sed -n 's,.*>\(.*\)</.*,\1,p')
+  fbHardw=$(echo "$jBoxInfo" | grep 'j:HW' | sed -n 's,.*>\(.*\)</.*,\1,p')
+  fbRev=$(echo "$jBoxInfo" | grep 'j:Revision' | sed -n 's,.*>\(.*\)</.*,\1,p')
+  fbSN=$(echo "$jBoxInfo" | grep 'j:Serial' | sed -n 's,.*>\(.*\)</.*,\1,p')
+  fbVersion1=$(echo "$jBoxInfo" | grep 'j:Version' | sed -n 's,.*>\(.*\)</.*,\1,p')
+  fbVersion2=$(echo "$fbVersion1" | sed -n 's/\(.*\)[.]\(.*\)[.]\(.*\)/\2.\3/p')
+
+# Login und Abruf der benötigten Seiten
+# Login and retrieve the required pages 
+  LOGIN
+  _PAGE_DSL_STATS=$($WEBCLIENT "$FritzBoxURL/internet/dsl_stats_tab.lua?sid=$SID")
+  _PAGE_INET_STAT=$($WEBCLIENT "$FritzBoxURL/internet/inetstat_monitor.lua?sid=$SID")
+  _PAGE_INET_COUN=$($WEBCLIENT "$FritzBoxURL/internet/inetstat_counter.lua?sid=$SID")
+  echo $_PAGE_DSL_STATS > /var/log/fritzbox7490.txt
+
+# Verarbeitung der Daten der Seite Online-Monitor 
+# Processing the data of the Online Monitor page 
+  _AKT_IP=$(echo "$_PAGE_INET_STAT" | sed -n 's/.*<br>IP-Adresse: \(.*\)<\/span>.*/\1/p')
+  _VERBAB=$(echo "$_PAGE_INET_STAT" | sed -n 's/.*<span class="limited">verbunden seit \(.*\)<\/span>,.*/\1/p')
+
+# Verarbeitung der Daten der Seite Online-Zähler
+# Processing the data of the page Online counter 
+  _ONLINE_Zeit=$(echo "${_PAGE_INET_COUN}" | tr -d "\n" | awk 'match($0,/Heute<\/td>.*"time">([0-9,:]+)<\/td>.*Gestern/,arr){print arr[1]};')
```

Comment:
Probably these repetitive filters can be moved into a function in order to generate more readable code.